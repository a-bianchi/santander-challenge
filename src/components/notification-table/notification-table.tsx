import { makeStyles, Theme } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import { useTranslation } from 'react-i18next';

const useStyles = makeStyles((theme: Theme) => ({
    table: {
        minWidth: 650,
    },
    divider: {
        marginTop: theme.spacing(4),
    },
    createButton: {
        display: 'flex',
        justifyContent: 'right',
        marginBottom: theme.spacing(1),
    },
    handleButton: {
        marginLeft: theme.spacing(1),
    },
}));

export type DataTableNotificationType = {
    message: string;
    date: string;
};

interface Props {
    title: string;
    data?: DataTableNotificationType[] | [];
}

export const NotificationTable = ({ title, data = [] }: Props): JSX.Element => {
    const classes = useStyles();
    const { t } = useTranslation();
    return (
        <div className={classes.divider}>
            <h1 data-test="Title">{title}</h1>
            <TableContainer component={Paper}>
                <Table data-test="TableNotification" className={classes.table} aria-label="simple table">
                    <TableHead data-test="Header">
                        <TableRow>
                            <TableCell align="center">{t('dataTable.message')}</TableCell>
                            <TableCell align="center">{t('dataTable.date')}</TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody data-test="BodyTableNotification">
                        {data.map((row, index) => (
                            <TableRow key={`${row.date}-${index}`}>
                                <TableCell align="center">{row.message}</TableCell>
                                <TableCell align="center">{row.date}</TableCell>
                            </TableRow>
                        ))}
                    </TableBody>
                </Table>
            </TableContainer>
        </div>
    );
};
