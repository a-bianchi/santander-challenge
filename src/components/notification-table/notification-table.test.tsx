/* global describe expect it */
import { shallow } from 'enzyme';
import { NotificationTable } from './notification-table';

describe('Notification Table', () => {
    it('should be defined', () => {
        expect(NotificationTable).toBeDefined();
    });
    it('should render correctly', () => {
        const tree = shallow(<NotificationTable title={'title'} />);
        expect(tree.find('.Table')).toBeDefined();
        expect(tree.find('.Header')).toBeDefined();
        expect(tree.find('.Body')).toBeDefined();
        expect(tree).toMatchSnapshot();
    });
    it('should render table', () => {
        const tree = shallow(
            <NotificationTable title={'title'} data={[{ message: 'some message', date: '04-04-2021' }]} />,
        );
        expect(tree.find('.title')).toBeDefined();
        expect(tree.find('.TableNotification')).toBeDefined();
        expect(tree).toMatchSnapshot();
    });
});
