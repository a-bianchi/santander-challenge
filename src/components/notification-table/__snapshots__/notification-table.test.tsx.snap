// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Table should render correctly 1`] = `ShallowWrapper {}`;

exports[`Notification Table should render table 1`] = `ShallowWrapper {}`;
